### PR TITLE
fix: string escape for fwd slash

### PR DESCRIPTION
**Pull request by mgrip**
#176

### CODE REVIEW

File: tests/string/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -85,14 +87,14 @@ $string = "foo\\\\\\\\bar";
...
 $string = "foo'bar";
-$string = "foo\\377bar";
-$string = "foo\\3";
-$string = "foo\\xFFbar";
-$string = "foo\\xr";
-$string = "foo\\x";
-$string = "foo\\u{ff}bar";
-$string = "foo\\ubar";
-$string = "foo\\u";
+$string = "foo\\\\377bar";
```
Comment (evilebottnawi) [inline tests/string/__snapshots__/jsfmt.spec.js.snap]: Broke :disappointed:

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: Yup my bad, didn't realize what these were - just put back!

File: tests/string/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -57,15 +57,17 @@ $string = "This is the value of the var named by the return value of getName():
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 <?php
 $string = 'this is a simple string';
 // $string = 'You can also have embedded\\n newlines in
 // strings this way as it is
 // okay to do';
 $string = 'Arnold once said: "I\\'ll be back"';
-$string = 'You deleted C:\\*.*?';
-$string = 'You deleted C:\\*.*?';
+$string = 'You deleted C:\\\\*.*?';
```
Comment (evilebottnawi) [inline tests/string/__snapshots__/jsfmt.spec.js.snap]: How about `\\\'`?

File: tests/string/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: @mgrip do not need to admit mistakes, it's okay search better solution

File: tests/string/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: Ya im confused - the test is

```php
$string = 'You deleted C:\\*.*?';
$string = 'You deleted C:\*.*?';
```

So I'm not sure why these are the same (in either case)... The snapshot escaping just makes this so much more confusing haha

Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: and `\`, `\\`, `\\\` just for verification tests

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: will do 👍

Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: $string = 'You deleted C:\\*.*?'; - just two slash
$string = 'You deleted C:\*.*?'; - just one slash
In snapshot each slash should be escaped

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: err actually `\` and `\\\` are invalid because they escape the closing quote

Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: @mgrip when I think that something is wrong or wrong, i just create `test.php` with content and `echo $string;` and look how it is output :smile:

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: right right - so snapshot should be
```php
$string = 'You deleted C:\\\\*.*?';
$string = 'You deleted C:\\*.*?';
```
right?

I'm just not sure why its incorrect atm. They're both the same

Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: @mgrip this examples was got from `php.net` :smile:

Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: yep it is right, first with two slashes, second with one slash

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: Ok its a parser error - I'll open a new issue

image

**Comment by mgrip** [comment]
@evilebottnawi ignoring the parser issue I think this should be good now!

**Pull request merged**